README.md

```diff
@@ -124,6 +125,20 @@ Key:
 * `=` Maintenance is manual, only severe security bugs will be patched.
 * `-` Deprecated; please upgrade.
 
+#### Dependency Matrix
+Since there are multiple libraries that client-go is either required to, or can work
+with, this is a list of all the libraries and versions that work with each other.
+
+| client-go                         | [apimachinery](https://github.com/kubernetes/apimachinery) | [apiextensions-apiserver](https://github.com/kubernetes/apiextensions-apiserver)<sup>*</sup> | [code-generator](https://github.com/kubernetes/code-generator)<sup>+</sup> |
+| ----------------------------------| -----------------------------------------------------------|----------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
+| tag: v2.0.x / branch: release-2.0 | -                                                          | -                                                                                            | -                                                                          |
+| tag: v3.0.x / branch: release-3.0 | branch: release-1.6                                        | -                                                                                            | -                                                                          |
+| tag: v4.0.x / branch: release-4.0 | branch: release-1.7                                        | branch: release-1.7                                                                          | branch: release-1.7                                                        |
+| tag: v5.0.x / branch: release-5.0 | branch: release-1.8                                        | branch: release-1.8                                                                          | branch: release-1.8                                                        |
```

Comment:
we do not support v5.0.x with release-1.8 in general. We support branch HEADs together, and we support v5.0.x with kubernetes-1.8.x (note: the same `x`).

Reply:
I have to admit - this is very confusing.

What is the point of the release branches then in all the repositories? Do they serve a purpose?
(This is why I made the PR, because this is not clear)

To make a more positive statement - would it be correct to say that (for example):

tag v.5.0.x  in client-go, will be supported with tag kubernetes-1.8.x in apimachinery (and other repositories as well)

Further question: does the x need to line up? or does it not matter?

Reply:
Deleted previous comment after reading what you wrote 3 times. This is _very_ confusing.

I'll rewrite the table, see if I can capture what you are saying.

Reply:
You sentence in the previous comment was good. v5.0.x with kubernetes-1.8.x of the other repos. v6.0.y with kubernetes-1.9.

If you talk about branches, release-5.0 HEAD (i.e. the current top commit) matches release-1.8 HEAD.

Reply:
Cool - thanks for explaining.

The only question I have (and I'd like to capture this) - is what you feel the reason why someone would use the version tag vs using the release branch?

Do we want to push users to using one over the other?

Reply:
To make matters worse https://github.com/kubernetes/apimachinery doesn't have any 1.6.x or 1.7.x tags, it only has a release-1.6 and release-1.7 branches.

So for older versions - use the release branches, but for newer ones use the tags?

Reply:
I actually realised this is even more confusing.

client-go also has kubernetes-1.8.x tags... which don't line up with the client go 5.0.x tags

Should we be pushing people towards that?